softwareComponents/configuration/json_include/configuration/serialization.hpp

```diff
@@ -331,22 +331,19 @@ namespace rofi::configuration::serialization {
         res[ "spaceJoints"  ] = json::array();
 
         for ( const auto& m : bot.modules() ) {
-            json j;
-            switch ( m.module->type ) {
-                case ModuleType::Universal:
-                    j = details::moduleToJSON( dynamic_cast< const UniversalModule& >( *m.module ), attrCb );
-                    break;
-                case ModuleType::Pad:
-                    j = details::moduleToJSON( dynamic_cast< const Pad& >( *m.module ), attrCb );
-                    break;
-                case ModuleType::Unknown:
-                    j = details::moduleToJSON( dynamic_cast< const UnknownModule& >( *m.module ), attrCb );
-                    break;
-                default:
-                    ROFI_UNREACHABLE( "Unknown type of a module" );
-            }
+            auto get_module_by_type = []( const Module& mod, Callback& cb ){
+                switch ( mod.type ) {
+                    case ModuleType::Universal:
+                        return details::moduleToJSON( dynamic_cast< const UniversalModule& >( mod ), cb );
+                    case ModuleType::Pad:
+                        return details::moduleToJSON( dynamic_cast< const Pad& >( mod ), cb );
+                    case ModuleType::Unknown:
+                        return details::moduleToJSON( dynamic_cast< const UnknownModule& >( mod ), cb );
+                }
+                ROFI_UNREACHABLE( "Unknown type of a module" );
+            };
```

Comment:
Tak pak je otázka, jestli to nemá být standalone funkce. Ale nechám na tobě, nemám silný názor.

Reply:
Souhlasím... taky nemám silný názor. Přikláněl jsem se k tomu to měnit co nejmíň (ale chci aby mi to vyhodilo warning, když se přidá další case).